core/src/main/scala/cromwell/util/DatabaseUtil.scala

```diff
@@ -0,0 +1,22 @@
+package cromwell.util
+
+import java.sql.SQLTransientException
+
+import akka.actor.ActorSystem
+import cromwell.core.retry.{Retry, SimpleExponentialBackoff}
+
+import scala.concurrent.Future
+import scala.concurrent.duration._
+import scala.language.postfixOps
+
+object DatabaseUtil {
+  private def isTransient(throwable: Throwable): Boolean = throwable match {
+    case _: SQLTransientException => true
+    case _ => false
+  }
+
+  def withRetry[A](f: => Future[A])(implicit actorSystem: ActorSystem): Future[A] = {
```

Comment:
In this example the `Future` actually runs only once, despite it being passed as a thunk:

```scala
scala> val f = Future {
     |   println("aw crap")
     |   throw new RuntimeException
     | }
aw crap
f: scala.concurrent.Future[Nothing] = scala.concurrent.impl.Promise$DefaultPromise@490dcafc

scala> def retry(r: => Future[Nothing]) {
     |   r
     |   r
     | }
retry: (r: => scala.concurrent.Future[Nothing])Unit

scala> retry(f)

scala> 
```